README.md

```diff
@@ -1,134 +1,117 @@
 
 ```mjs
-import a from './a.mjs'
-import b from './b.mjs'
-import c from './c.mjs'
-
-console.log(a, b, c);
+// awaiting.mjs
+import { process } from "./some-module.mjs";
+let output;
+export default (async () => {
+  const dynamic = await import(computedModuleSpecifier);
+  const data = await fetch(url);
+  output = process(dynamic.default, data);
+})();
+export output;
 ```
 
-If each of the modules above had a top level await present the loading would have similar execution order to
+Then, the module could be used as:
 
 ```mjs
-(async () => {
-  const a = await import('./a.mjs');
-  const b = await import('./b.mjs');
-  const c = await import('./c.mjs');
-  
-  console.log(a, b, c);
-})();
+// usage.mjs
+import default, { output } from "./awaiting.mjs";
+export function outputPlusValue(value) { return output + value }
+
+default.then(() => {
+  console.log(outputPlusValue(100));
+  setTimeout(() => console.log(outputPlusValue(100), 1000)
+});
 ```
 
-Module a would need to finish executing before b or c could execute.
+However, this leaves us with a number of problems still:
+- Everyone has to learn about a particular protocol to find the right Promise to wait on the module being loaded
+- If you forget to apply the protocol, things might "just work" some of the time (due to the race being won in a certain way)
+- In a deep module hierarchy, the Promise needs to be explicitly threaded through each step of the chain.
 
-### Variant B: top-level `await` does not block sibling execution
+For example, here, we waited on the promise from `"./awaiting.mjs"` properly, but we forgot to re-export it, so modules that use our module may still run into the original race condition.
 
-In this proposed solution a call to `top-level await` would block execution of parent nodes in the graph but would allow siblings to continue to execute. 
+### Solution: Top-level `await`
 
-In this implementation you could consider the following
+Top-level `await` lets us rely on the module system itself to handle all of these promises, and make sure that things are well-coordinated. The above example could be simply written and used as follows:
 
 ```mjs
-import a from './a.mjs'
-import b from './b.mjs'
-import c from './c.mjs'
-
-console.log(a, b, c);
+// awaiting.mjs
+import { process } from "./some-module.mjs";
+const dynamic = await import(computedModuleSpecifier);
+const data = await fetch(url);
+export const output = process(dynamic.default, data);
+
+// usage.mjs
+import { output } from "./awaiting.mjs";
+export function outputPlusValue(value) { return output + value }
+
+console.log(outputPlusValue(100));
+setTimeout(() => console.log(outputPlusValue(100), 1000)
 ```
 
-If each of the modules above had a top level await present the loading would have similar execution order to
+None of the statements in `usage.mjs` will execute until the `await`s in `awaiting.mjs` have had their Promises resolve, so the race condition is avoided by design.
```

Comment:
You could add something like "This is similar to [an extension of?] how in the current system none of the statements in usage.mjs will execute until `awaiting.mjs` is loaded and all of its statements has executed."